src/node_sqlite.h

```diff
@@ -159,6 +172,10 @@ class DatabaseSync : public BaseObject {
   void SetIgnoreNextSQLiteError(bool ignore);
   bool ShouldIgnoreSQLiteError();
 
+  bool HasPendingAuthorizerError() const;
+  void StoreAuthorizerError(v8::Local<v8::Value> error);
+  void RethrowPendingAuthorizerError();
```

Comment:
Actually, just checking – @geeksilva97's PR at #60028 just made me aware of `SetIgnoreNextSQLiteError()`, that seems to be a better API than this + `TryCatch` ... is there a reason we couldn't use that instead?

`TryCatch` is primarily useful if you need to store an exception across an asynchronous context, but I think everything here is fully synchronous

Reply:
I’ve attempted this, but it didn’t work, and I couldn’t comprehend why. Throwing exceptions from the `AuthorizerCallback` doesn’t result in any errors in JavaScript.

```cpp
  if (try_catch.HasCaught()) {
    isolate->ThrowException(try_catch.Exception());
    db->SetIgnoreNextSQLiteError(true);
    return SQLITE_DENY;
  }
```

Reply:
You can probably remove this `try_catch` and the `ThrowException` thing. [If `Call` throws, an exception will be pending](https://github.com/nodejs/node/pull/56600#discussion_r2008838925)